### PR TITLE
Really fix it this time spectral library generation: with updated mzlib; updated library generation

### PR DESCRIPTION
**Pull request by Yuling1996**

### CODE REVIEW

**Review by trishorts — APPROVED** [review]
this PR looks good to me. i can see a few ways that it can be improved which i have explained to yuling, but these are not critical issues. the code functions well, both creating and using a spectral library.

File: TaskLayer/SearchTask/PostSearchAnalysisTask.cs [hunk 1]
```diff
@@ -580,6 +585,35 @@ private void WritePsmResults()
...
+                
+                if (!PsmsGroupByPeptideAndCharge.ContainsKey(peptideWithChargeState))
+                {
+                    PsmsGroupByPeptideAndCharge.Add(peptideWithChargeState, psmsWithSamePeptideAndSameCharge);
+                }
+            }
+            var spectraLibrary = new List<LibrarySpectrum>();
+            foreach(var psm in PsmsGroupByPeptideAndCharge)
+            {
+                var standardSpectrum = new LibrarySpectrum(psm.Value[0].FullSequence, psm.Value[0].ScanPrecursorMonoisotopicPeakMz, psm.Value[0].ScanPrecursorCharge, psm.Value[0].MatchedFragmentIons, psm.Value[0].ScanRetentionTime);
```
Comment (nbollis) [inline TaskLayer/SearchTask/PostSearchAnalysisTask.cs:610]: Does using retention time as a parameter assume identical LC conditions?

File: TaskLayer/SearchTask/PostSearchAnalysisTask.cs [hunk 2]
```diff
@@ -580,6 +585,35 @@ private void WritePsmResults()
...
+        //for those spectra matching the same peptide/protein with same charge, save the one with highest score
+        private void SpectralLibraryGeneration()
+        {
+            var FilteredPsmList = Parameters.AllPsms
+               .Where(p => p.FdrInfo.PEP_QValue <= 0.01 && p.FdrInfo.QValueNotch <= CommonParameters.QValueOutputFilter).ToList();
+            FilteredPsmList.RemoveAll(b => b.IsDecoy);
+            FilteredPsmList.RemoveAll(b => b.IsContaminant);
+            //group psms by peptide and charge, the psms having same sequence and same charge will be in the same group
+            Dictionary<(String, int), List<PeptideSpectralMatch>> PsmsGroupByPeptideAndCharge = new Dictionary<(String, int), List<PeptideSpectralMatch>>();
+            foreach (var x in FilteredPsmList)
```
Comment (nbollis) [inline TaskLayer/SearchTask/PostSearchAnalysisTask.cs:597]: Could a more clear variable be used here instead of x

**Review by nbollis — APPROVED** [review]
Looks good. Left one minor comment about variable naming

**Review by Alexander-Sol — APPROVED** [review]
Looks good + all tests pass.

File: TaskLayer/SearchTask/PostSearchAnalysisTask.cs (hunk 2 above)
Comment (Yuling1996) [reply TaskLayer/SearchTask/PostSearchAnalysisTask.cs:597]: Thanks for the advice!

**Review by rmmiller22 — APPROVED** [review]

**Pull request merged**